internal/common/predicate.go

```diff
@@ -12,3 +14,14 @@ func IsControlledBy(object client.Object, owner *metav1.OwnerReference) bool {
 		controller.APIVersion == owner.APIVersion &&
 		controller.Kind == owner.Kind
 }
+
+// FetchCurrentNamespaceName returns namespace name by using information stored as file
+// Returns default Openshift Builds namespace on error
+// Refer: https://kubernetes.io/docs/tasks/run-application/access-api-from-pod/#without-using-a-proxy
+func FetchCurrentNamespaceName() string {
+	content, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
```

Comment:
Per the linked documentation:

```
Finally, the default namespace to be used for namespaced API operations is placed in a file at /var/run/secrets/kubernetes.io/serviceaccount/namespace in each container.
```

I _assume_ this is the namespace the pod is deployed in?